app/rankers/algorithms.py

```diff
@@ -112,18 +112,27 @@ def thompson_sampling(
     return [x[0] for x in scores]
 
 
-async def personalize_topic_slates(input_slate_configs: List['SlateConfigModel'],
-                                   personalized_topics: PersonalizedTopicList) -> List['SlateConfigModel']:
+def personalize_topic_slates(input_slate_configs: List['SlateConfigModel'],
+                             personalized_topics: PersonalizedTopicList,
+                             topic_limit: Optional[int] = None) -> List['SlateConfigModel']:
```

Comment:
This parameter is currently unused, so I suggest we remove it.

It's implemented as a simple limit before returning the result, so it could be replaced by a top_n_ ranker, which I think would be a cleaner solution, because it would keep our rankers focused. I can see how there could be more complex constraints (e.g. remove all but the first X topic slates), but let's cross that bridge when the need is there.
```suggestion
                             ) -> List['SlateConfigModel']:
```

Reply:
I think this allows for more flexibility in the absence of an actual top N ranker.  right now we have 5, 15, and 30.  I think android already wants top1 and web wants top 3.